### ISSUE TITLE
Broken Splash Texts

### ISSUE THREAD

**Issue by Julienraptor01** [LudoCrypt/Vistas#25]
The only splash text displayed is missingno unless the Random Per Screen option is enabled

From discord
> My first problem is simple
If Random Per Screen isn't enabled the splash text is missingno (I'm not using any resource pack or other mod just barebones install)
But if the option is enabled then a real splash text appear
And once one as appeared even if i toggle the option back to disabled the splash won't change back to missingno until a restart of the game
I've tried to change the panorama.json from vistas (the one inside the mod) to only make it default with nothing in it but it doesn't change anything
I've even tried to delete the panorama.json and it doesn't change anything too except the fact even with the Random Per Screen enabled doesn't fix the problem anymore but i assume this is because the mod doesn't load any splash texts at all anymore

**Comment by osfanbuff63** [comment]
What's wrong with enabling random per screen? I assume they don't even have Mod Menu either. Personally, I think random per screen should be enabled by default, given 1) the primary features of this mod are through that and 2) it doesn't cause missingno on startup.

**Comment by Julienraptor01** [comment]
> What's wrong with enabling random per screen? I assume they don't even have Mod Menu either. Personally, I think random per screen should be enabled by default, given 1) the primary features of this mod are through that and 2) it doesn't cause missingno on startup.

Nothing is wrong with it as it fixes the bug but having it disabled make the splash text to be always missingo
I'll add more information about my versions later

**Comment by gniftygnome** [comment]
I think one way to resolve this is the following patch (which I have added to my Vistas 1.18.2 and 1.19 PRs):

```java
	private void vistas$get(CallbackInfoReturnable<String> ci) {
		MinecraftClient client = MinecraftClient.getInstance();
		PanoramaResourceReloader resourceReloader = ((MinecraftClientAccess) client).getPanoramaResourceReloader();
+		Identifier panoramaId = VistasTitle.PANORAMAS_INVERT.get(VistasTitle.CURRENT.getValue());
+
+		if (resourceReloader != null && panoramaId != null) {
-		if (resourceReloader != null) {
			ci.setReturnValue(resourceReloader.get());
		}
	}
```

The idea is because the Vistas PanoramaResourceReloader will return a hard-coded splash text of "missingno" in the event the following test fails:  `if (panoramaId != null)` we should try to avoid using the Vistas PanoramaResourceReloader to set the text via the SplashTextResourceSupplier mixin when the test fails.  Practically speaking what this means is to allow vanilla to set the splash text when vanilla is setting the panorama.

**Comment by gniftygnome** [comment]
This was resolved in versions 2.2.0 and 2.3.0; I am not certain it will get backported to older versions.

**Comment by Julienraptor01** [comment]
> This was resolved in versions 2.2.0 and 2.3.0; I am not certain it will get backported to older versions.

That's not a problem, i keep updating tothe lastest minecraft version, thank you
